pkgs/applications/editors/vscode/extensions/default.nix

```diff
@@ -4,7 +4,7 @@
 , callPackage
 , vscode-utils
 , asciidoctor
-, nodePackages
+, pyright
```

Comment:
```suggestion
```
- Please split commit for `vscode-extension`.
- As a new parameter is necessary, it is ideal to split the extension to it's own directory to avoid polluting the parameters list. (That is going to be eventually cleaned-up on upgrades.)

Reply:
Did the commit split, as for extension split I don't use vscode so that would require me to learn its ecosystem first I don't have time for that, sorry.

Reply:
Please don't mark issue as 'resolved' while I haven't provided feedback.

Just:

1) Use of callPackage to call it:

> ms-python.vscode-pylance = callPackage ./ms-python.vscode-pylance { };

2) Copy and paste of extension body to a discrete file:

See [example](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/extensions/equinusocio.vsc-material-theme/default.nix).

It's simple.